### DISCUSSION TITLE
Introduce yourself

### DISCUSSION THREAD

**Discussion by boris-kz** [boris-kz/CogAlg#61] [General]
If you feel like it :)

**Comment by kwcckw** [comment]
Hello guys , I'm Chee from Malaysia. I'm from electronic engineering background and i've been working on image processing & computer vision related stuff for around 8 years. I'm very interested to see the initial prototype version of CogAlg and hopefully we are able to reach that stage soon.

**Reply by boris-kz** [reply]
I need to add that Chee was a lead coder of this project for 1.5 years. Without him, this project would be far behind. Thank you very much Chee!

**Comment by alex-pitertsev** [comment]
Hello everybody. I am Alex from Ukraine. I am a Ph.D., currently freelancing and teaching some students at university the basics of electronics, image and signal processing in Python. I am interested in helping develop a new approach in the field of machine learning.

**Comment by eswistak88** [comment]
Hello All,

I am Ethan. Just got introduced to the project via Chris.

I am a former accountant and aspiring AI researcher currently finishing up my Master's(in Cognitive Systems) at the Universitat Ulm in Germany. After running some basic neural networks and teaching a class in Convolutional neural networks I got interested in some alternative ideas and came across Numenta and also the spinoff by Jeff Hawkins protege Vicarious. I also got mixed up working on some projects for some PhD's in various departments.

I do find it interesting that CogAlg is not looking at a biological basis or a statistical  basis for intelligence but instead kind of seems to be looking to refine what intelligence really IS at it's core. I actually have a few ideas about that, not sure if they're relevant but maybe there's some connection I'm not seeing.

The way I see it, all these different machine learning algorithms and the human brain really only seeks to do one thing, which is minimize entropy. Of course, the easiest way to minimize entropy is just to make something completely uniform, but I believe the universe has this natural tendency towards complexity, so you can't really get away with that. The predictions and the patterns are really just a byproduct of that process and there are certainly many different functions that will be able to minimize entropy, not just one universal one. As a human, you know, you'd like everything to stay the same but you feel pain, you get hungry, tired, stuff changes essentially so the best your brain can do to minimize that entropy is predict and then take actions to avoid anything unexpected happening.

Anyway, that does actually bring me to a useful question perhaps. How exactly does this CogAlg structure itself such that the entropy of the system is minimized? I've read a lot about it but I don't think I saw that mentioned.

Hope I'm asking this question right. I'd consider myself a better coder than a mathematician really.

**Reply by boris-kz** [reply]
Hi Ethan, welcome! 
Info-theoretical entropy is a complimentary of compressibility, and this algorithm is being designed to maximize projected match: lossless component of compression. This is in my "Atomic comparison.." section. But please note that it's actually a very significant "expansion" within a level, effective compression (selective processing) is only supposed to happen on the next level. So the alg is supposed increase compression / reduce entropy of higher-level representations w.r.t. raw input, real and projected. 

That is completely separate from any specific values you can add via RL or supervision, or evolution adds as conditioning or instincts.
Not directly related, but I discuss these things on ontological level in my old blog:  https://meta-evolution.blogspot.com/. If you are curious, I think that intuitively understood physical entropy growth is actually a reduction of info-theoretical entropy: https://meta-evolution.blogspot.com/2012/01/entropy-growth-as-increase-of-order.html.

**Reply by ASBJ021** [reply]
Hello Everyone,

I am Akash. I am Automation & Robotics engineer who graduated in 2020 from KLE Tech University, INDIA.  Currently, I am working as a freelancer.  I am working with computer vision, python, and ML for the last 3 years and still learning new things.
I came to know about CogAlg when Chris contacted me. Then I went through this repo and try to understand the concepts and the code.  I am looking forward to work with you guys on this algorithm.  

Thank you.

**Comment by TufailAhmadSiddiq** [comment]
Hello All,

My name is Tufail Ahmad Siddiq. I have completed my bachelor's degree in Electrical Engineering with a specialization in Computing from Namal Institute Mianwali, Pakistan in 2020. I have been working as a full-time freelancer and offer my services in Machine Learning, Computer Vision, and Data Science. I have been shared this repository with Chris to study. I have studied the repository and made a sketch in my mind.

According to the information given in the READ section, I have figured out that you are trying to create an algorithm contains properties of several existing deep learning algorithm for application of Computer Vision. You have mentioned a cognitive system. So, I have a few questions. I am not sure these are related to the objective of this project. In the cognitive system, the system utilizes information (data) to help users make smart decisions. This is done through some data analysis and a pool of good options is kept in front of a user. But in an AI system, the system makes the decision itself on the basis of its learning using minimizing the entropy as Ethan said. So, my question is regarding the architectural objective of this project. What is being tried to achieve in the project?

**Reply by boris-kz** [reply]
Hi Tufail, appreciate your interest. As I tried to explain in the first paragraph, the objective is pattern discovery and prediction. Decisions will be an organic feedback, to maximize predictive power of the system. If that sounds too abstract, this project is not for you.

**Reply by TufailAhmadSiddiq** [reply]
I understand feedback and a predictive system. I get interested in this project due to designing a new algorithm containing properties of other neural networks. I am willing to work with you.

**Reply by boris-kz** [reply]
Great, do  you have more specific questions? Please note that design is totally different from ANN, see comparison to ANN and BNN section. Our plan is explained in wiki: https://github.com/boris-kz/CogAlg/wiki, currently working on line_PPs: https://github.com/boris-kz/CogAlg/blob/master/line_1D_alg/line_PPs.py, but first you have to understand line_patterns: https://github.com/boris-kz/CogAlg/blob/master/line_1D_alg/line_patterns.py

**Reply by TufailAhmadSiddiq** [reply]
Okay, I will go through the plan again. I will study line_patterns.py and understand it. I will join you when I understand the previous work.

**Reply by boris-kz** [reply]
Thanks!
